#!/usr/bin/env python
PACKAGE = "pyramid_evaluation"

from math import pi
from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("enable_disturbance", bool_t, 0, "enable_disturbance", False)

gen.add("force_x", double_t, 0, "force_x", 0.0, -5.0, 5.0)
gen.add("force_y", double_t, 0, "force_y", 0.0, -5.0, 5.0)
gen.add("force_z", double_t, 0, "force_z", 0.0, 0.0, 10.0)
gen.add("torque_x", double_t, 0, "torque_x", 0.0, -0.1, 0.1)
gen.add("torque_y", double_t, 0, "torque_y", 0.0, -0.1, 0.1)
gen.add("torque_z", double_t, 0, "torque_z", 0.0, -0.1, 0.1)

disturbance_modes = gen.enum([gen.const("step", str_t, "step", ""),
                              gen.const("sine_wave", str_t, "sine_wave", "")],
                              "Disturbance modes")

gen.add("disturbance_modes", str_t, 0, "Disturbance modes", "step", edit_method=disturbance_modes)

gen.add("semiMinorAxis", double_t, 0, "semiMinorAxis", 1.0, 0.0, 3.0)
gen.add("semiMajorAxis", double_t, 0, "semiMajorAxis", 1.0, 0.0, 3.0)

exit(gen.generate(PACKAGE, "pyramid_evaluation", "DisturbanceGenerator"))
